src/torchcodec/_core/_metadata.py

```diff
@@ -129,8 +129,15 @@ def num_frames(self) -> Optional[int]:
         """
```

Comment:
In the doc string, we should also say what we do if num_frames is missing from all of the metadata.